### PR TITLE
GEODE-4959: Add --if-not-exists option to create gateway receiver com…

### PR DESCRIPTION
**Pull request by PurelyApplied**
…mand.

* Command will now raise an error if a gateway receiver cannot be created on every member
* This error will *not* be raised if a gateway receiver already exists on a given member and --if-not-exists is provided.
* Cleanup test class.
* Add setStatus to CommandResult

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GatewayReceiverCreateFunction.java [hunk 1]
```diff
@@ -154,6 +165,10 @@ private Object newInstance(String className)
     return ClassPathLoader.getLatest().forName(className).newInstance();
   }
 
+  private boolean doesSomeGatewayReceiverAlreadyExists(Cache cache) {
```
Comment (pdxrunner) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GatewayReceiverCreateFunction.java]: Only one gateway-receiver is allowed per server, so the method name shouldn't suggest plurality. I suggest `gatewayReceiverExists`

Comment (PurelyApplied) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GatewayReceiverCreateFunction.java]: I'm bad at names.  Yours is better, thanks.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GatewayReceiverCreateFunction.java [hunk 2]
```diff
@@ -56,11 +56,28 @@ public void execute(FunctionContext context) {
     GatewayReceiverFunctionArgs gatewayReceiverCreateArgs =
         (GatewayReceiverFunctionArgs) context.getArguments();
 
+    // Exit early if a receiver already exists.
+    // Consider this a failure unless --if-not-exists was provided.
+    if (doesSomeGatewayReceiverAlreadyExists(cache)) {
+      CliFunctionResult result;
+      if (gatewayReceiverCreateArgs.getIfNotExists()) {
+        result = new CliFunctionResult(memberNameOrId, true,
+            "SKIP: A Gateway Receiver already exists on this member.");
```
[the remark this answers is not in the document]
Comment (PurelyApplied) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GatewayReceiverCreateFunction.java]: Done and done.

The same string also appears in `GatewayReceiverFactoryImpl`.  I can't decide if I dislike the idea of duplication across the files less than the spaghetti-cross-reaching between that class and the `*Function`.  Thoughts?

[the remark this answers is not in the document]
Comment (pdxrunner) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GatewayReceiverCreateFunction.java]: I have to say I'm in favor of the one string constant even if it involves reaching across the related classes. As there was been some discussion on the dev list that disfavors CliStrings, I've been putting string definitions like this in the appropriate `*Command` class

**Comment by PurelyApplied** [comment]
Stubbing my toe on test environment issues... Hopefully, this latest precheckin will actually finish, and I expect it to be green when it does.

**Comment by PurelyApplied** [comment]
precheckin green.

**Pull request merged**